src/__tests__/parser.test.ts

```diff
@@ -0,0 +1,457 @@
+      TWILIO_AUTH_TOKEN=
+    `;
+      const result = parse(file);
+      expect(result.variables).toEqual([
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_ACCOUNT_SID',
+          description: 'Your Twilio Account SID',
+          default: 'ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
+        },
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_AUTH_TOKEN',
+          description: 'Your Twilio Auth Token',
+          default: null,
+        },
+      ]);
+    });
+
+    test('recognizes format comment', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: Your Twilio Account SID
+      # format: sid
+      TWILIO_ACCOUNT_SID=ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      
+      # description: Your Twilio Auth Token
+      # format: secret
+      TWILIO_AUTH_TOKEN=
+    `;
+      const result = parse(file);
+      expect(result.variables).toEqual([
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_ACCOUNT_SID',
+          description: 'Your Twilio Account SID',
+          format: 'sid',
+          default: 'ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
+        },
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_AUTH_TOKEN',
+          description: 'Your Twilio Auth Token',
+          format: 'secret',
+          default: null,
+        },
+      ]);
+    });
+
+    test('handles invalid format values', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: Your Twilio Account SID
+      # format: boolean
+      TWILIO_ACCOUNT_SID=ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      
+      # description: Your Twilio Auth Token
+      # format: secret
+      TWILIO_AUTH_TOKEN=
+    `;
+
+      expect(() => parse(file)).toThrowError();
+    });
+
+    test('handles invalid list format values', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: Your Twilio Account SID
+      # format: list(boolean)
+      TWILIO_ACCOUNT_SID=ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      
+      # description: Your Twilio Auth Token
+      # format: secret
+      TWILIO_AUTH_TOKEN=
+    `;
+
+      expect(() => parse(file)).toThrowError();
+    });
+
+    test('handles invalid nested_list values', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: Your Twilio Account SID
+      # format: nest_list(phone_number,boolean)
+      TWILIO_ACCOUNT_SID=ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      
+      # description: Your Twilio Auth Token
+      # format: secret
+      TWILIO_AUTH_TOKEN=
+    `;
+
+      expect(() => parse(file)).toThrowError();
+    });
+
+    test('recognizes required comment', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: your Twilio phone number
+      # format: phone_number
+      # required: true
+      TWILIO_PHONE_NUMBER=+12223334444
+
+      # description: Message to play
+      # required: false
```

Comment:
what about an invalid case like `required: number`

Reply:
It throws an error. I added a test case for it